packages/experimental-utils/src/eslint-utils/applyDefault.ts

```diff
@@ -8,13 +8,13 @@ import { deepMerge, isObjectNotArray } from './deepMerge';
  * @returns the options with defaults
  */
 function applyDefault<TUser extends readonly unknown[], TDefault extends TUser>(
-  defaultOptions: TDefault,
-  userOptions: TUser | null,
+  defaultOptions: Readonly<TDefault>,
+  userOptions: Readonly<TUser> | null,
 ): TDefault {
   // clone defaults
-  const options: AsMutable<TDefault> = JSON.parse(
+  const options = JSON.parse(
     JSON.stringify(defaultOptions),
-  );
+  ) as AsMutable<TDefault>;
```

Comment:
hmmm - maybe it'd be good for us to add a flag to `no-unsafe-assignment`. Something like `allowAssignmentToAnnotatedVariableDeclaration` or something which would allow this case?

This seems like it's probably a pretty common thing in TS code.

Reply:
Talked a bit directly -- we like forcing people to `as` cast 😄